src/capturer/cosmos/cosmos.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ChangeFeedIteratorOptions,
+  ChangeFeedStartFrom,
+  Container,
+  StatusCodes,
+} from "@azure/cosmos";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import * as B from "fp-ts/boolean";
+import { pipe } from "fp-ts/lib/function";
+import { ContinuationTokenItem, upsertItem } from "./utils";
+
+export const getChangeFeedIteratorOptions = (
+  continuationToken?: string,
+  maxItemCount?: number
+): E.Either<Error, ChangeFeedIteratorOptions> =>
+  E.tryCatch(
+    () => ({
+      maxItemCount: maxItemCount && maxItemCount > 0 ? maxItemCount : 1,
+      changeFeedStartFrom: continuationToken
+        ? ChangeFeedStartFrom.Continuation(continuationToken)
+        : ChangeFeedStartFrom.Beginning(),
+    }),
+    E.toError
+  );
+
+export const processChangeFeed = (
+  changeFeedContainer: Container,
+  changeFeedIteratorOptions: ChangeFeedIteratorOptions,
+  leaseContainer: Container
+): TE.TaskEither<Error, void> =>
+  TE.tryCatch(async () => {
+    const changeFeedIterator = changeFeedContainer.items.getChangeFeedIterator(
+      changeFeedIteratorOptions
+    );
+    for await (const result of changeFeedIterator.getAsyncIterator()) {
+      pipe(
+        result.statusCode === StatusCodes.NotModified,
+        B.fold(
+          () =>
+            void upsertItem<ContinuationTokenItem>(leaseContainer, {
+              id: changeFeedContainer.id.replace(" ", "-"),
+              lease: result.continuationToken,
+            } as ContinuationTokenItem)(),
+          () => TE.right(void 0)
```

Comment:
Can we add a comment that allows to clarify which operation should be done in the right case (i.e. event hub publication) or even the desired behavior for each document read from change feed??